src/inc/til/size.h

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#ifdef UNIT_TESTING
+class SizeTests;
+#endif
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    class size
+    {
+    public:
+        constexpr size() noexcept :
+            size(0, 0)
+        {
+        }
+
+        // On 64-bit processors, int and ptrdiff_t are different fundamental types.
+        // On 32-bit processors, they're the same which makes this a double-definition
+        // with the `ptrdiff_t` one below.
+#if defined(_M_AMD64) || defined(_M_ARM64)
+        constexpr size(int width, int height) noexcept :
+            size(static_cast<ptrdiff_t>(width), static_cast<ptrdiff_t>(height))
+        {
+        }
+#endif
+
+        size(size_t width, size_t height)
+        {
+            THROW_HR_IF(E_ABORT, !base::MakeCheckedNum(width).AssignIfValid(&_width));
+            THROW_HR_IF(E_ABORT, !base::MakeCheckedNum(height).AssignIfValid(&_height));
+        }
+
+        constexpr size(ptrdiff_t width, ptrdiff_t height) noexcept :
+            _width(width),
+            _height(height)
+        {
+        }
+
+        template<typename TOther>

```

Comment:
nit: minor docs; "this is for converting from something with X, Y"

Reply:
done